appmanager.ipynb

```diff
@@ -103,12 +108,112 @@
     "        btn.on_click(lambda b: install_app(name))   \n",
     "        display(btn)\n",
     "    else:\n",
-    "        ver_sel = mk_version_selector(name)\n",
-    "        btn = ipw.Button(description=\"remove\")\n",
-    "        btn.on_click(lambda b: remove_app(name))\n",
-    "        display(ipw. HBox([btn, ver_sel]))\n",
+    "        \n",
+    "        btn_remove = ipw.Button(description=\"remove\")\n",
+    "        btn_remove.on_click(lambda b: remove_app(name))\n",
+    "        \n",
+    "        btn_check_updates = ipw.Button(description='check for updates')\n",
+    "        btn_check_updates.on_click(lambda b: check_single_update(name))\n",
+    "        \n",
+    "        ref_drop = ipw.Dropdown(description=\"Version:\")\n",
+    "\n",
+    "        display(ipw.HBox([btn_remove, btn_check_updates, ref_drop]))\n",
+    "\n",
+    "        ver_sel = mk_version_selector(name) # move this down so the UI is more responsive\n",
+    "        ref_drop.options=ver_sel['options']\n",
+    "        ref_drop.value=ver_sel['head']\n",
+    "        def on_ref_change(c):\n",
+    "            change_ref(name, ref_drop.value)\n",
+    "        ref_drop.observe(on_ref_change, names='value')\n",
+    "        \n",
+    "    try:\n",
+    "        btn_remove.disabled = name == \"home\"\n",
+    "    except UnboundLocalError:\n",
```

Comment:
When/why is this UnboundLocalError thrown?

Reply:
It should not be thrown at all now.